### ISSUE TITLE
ParamsLoader does not modify $_SERVER and $_ENV

### ISSUE THREAD

**Issue by benelori** [Codeception/Codeception#6394]
#### What are you trying to achieve?

I am trying to load params from a dotenv file

#### What do you get instead?

Because the params loading has changed to use just the `ArrayLoader`, our tests are failing, because `$_ENV` and `$_SERVER` are not updated.

I found the related issue: https://github.com/Codeception/Codeception/pull/6348

But I don't think this was a good idea, because any setups that rely on this side-effect will fail. e.g. Symfony project that loads its container params from environment variables.

### Details

* Codeception version: 5.0.0-RC1
* PHP Version:  PHP 8.0.11 
* Operating System: Debian
* Installation type: Composer
* List of installed packages (`composer show`)
* Suite configuration:

**Comment by benelori** [comment]
Related PR: https://github.com/Codeception/Codeception/pull/6393

**Comment by Naktibalda** [comment]
I noticed the same issue in Symfony tests and wrote about it in #6348

@mdoelker What do you think about restoring original behaviour?

**Comment by Naktibalda** [comment]
Merged

**Comment by mdoelker** [comment]
@Naktibalda just saw your comment (it got lost in the tons of emails raining down on me).

My take is that we are looking at two different use cases here. One is loading params from files that we can use in our config, which Codeception supports different input formats for. One of these formats happen to be env files. It's just another format with keys and values. When I load my params from a JSON or XML file, it doesn't update my environment as a side-effect, so why should that be the case when I load an `.env` file? It might be desirable to have your test framework do that for you as well, you might have different opinions on that. It is another use case though, and it should be explicit. Not as a byproduct of param loading.

My take is that it is *not* the responsibility of the test framework to change your env. If you rely on that, you are free to load and update your `$_ENV`/`getenv` from within your suite's bootstrap file using one of the dotenv libs or even your own code, the same way you would do it in your application. Nothing is preventing you to do so.

Reverting this reintroduces the buggy behavior that prevents you to mix and match param loading sources and leads to unexpected behavior.

I can do the following, and it works just fine:

```yml
params:
  - params.xml
  - more-params.json # overrides values in params.xml
```

```yml
params:
  - more-params.json
  - params.xml # overrides values in more-params.json
```

```yml
params:
  - params.xml
  - more-params.json # overrides values in params.xml
  - .env # overrides values in params.xml and more-params.json
```

But here we get problems:

```yml
params:
  - env # use current env vars as params
  - .env # override with values from .env file
```

```yml
params:
  - .env # use values from .env files
  - env # but let REAL env values take precedence.... wait... this doesn'nt work, because the .env line above already messed up the env
```

Also, if you should happen to load an .env file for params but need the REAL env values in your tests, that may be defined in your pipeline or container or wherever, then you're lost as well.

That's why I don't think reverting this is a good idea. What we should do instead is explain how you can achieve the same behavior in case you rely on it using boostrap code.

**Comment by Naktibalda** [comment]
I agree that application code depends on unexpected side-effect of the test framework and it would be better to move environment loading to framework module.

There is a known issue that .env file loading of Symfony applications is incorrect https://github.com/Codeception/module-symfony/issues/99 , but it is better to have partially incorrect behaviour than completely incorrect behaviour.

We can revert the revert after fixing Symfony module.
I could use some help, because I haven't used Symfony with Codeception for 2 years.

@benelori Could you confirm that you are using module-symfony too?

**Comment by benelori** [comment]
Yes, I am using module-symfony. And I can also try to tackle the issue, if that's ok.

If this feature is moved to module, then wouldn't we need to do this for other framework specific modules? Wouldn't it be ok to just offer documentation on how to do this in a global bootstrap for example?
